commands/user.js

```diff
@@ -0,0 +1,132 @@
+	help: true,
+	checkPerm(perm, commandName) {
+		switch (commandName) {
+			case 'user':
+			case 'kick':
+				return perm >= global.PERM_RECRUITER;
+			case 'ban':
+				return perm >= global.PERM_MOD;
+		}
+		return false;
+	},
+	async execute(interaction, member, perm, permName) {
+		const subCommand = interaction.options.getSubcommand();
+		let targetMember = interaction.options.getMember('user');
+
+		if (targetMember) {
+			// Validate caller has permissions to kick/ban target if they are a member of the server
+			let [targetPerm, targetPermName] = global.getPermissionLevelForMember(targetMember);
+			if (!targetMember.kickable || perm <= targetPerm)
+				return global.ephemeralReply(interaction, `You do not have permissions to ${subCommand} ${targetMember}.`);
+		}
+
+		switch (subCommand) {
+			case 'kick': {
+				if (!targetMember)
+					return global.ephemeralReply(interaction, 'User is invalid or left the server.');
+
+				let reason = interaction.options.getString('reason') ?? "No reason provided";
+
+				const confirm = new ButtonBuilder()
+					.setCustomId('confirm_user_kick')
+					.setLabel('Confirm Kick')
+					.setStyle(ButtonStyle.Danger);
+				const cancel = new ButtonBuilder()
+					.setCustomId('cancel_user_kick')
+					.setLabel('Cancel')
+					.setStyle(ButtonStyle.Secondary);
+				const row = new ActionRowBuilder()
+					.addComponents(cancel, confirm);
+				const response = await global.ephemeralReply(interaction, {
+					content: `Are you sure you want to kick ${targetMember} from the server?`,
+					components: [row]
+				});
+
+				const filter = (i) => (i.customId === 'confirm_user_kick' || i.customId === 'cancel_user_kick') && i.user.id === interaction.user.id;
+				try {
+					const confirmation = await response.awaitMessageComponent({ filter: filter, time: 10000 });
+					if (confirmation.customId === 'confirm_user_kick') {
+						await targetMember.kick(`Requested by ${global.getNameFromMessage(interaction)}: ${reason}`)
+							.catch(error => global.ephemeralReply(interaction, `Sorry, I couldn't kick because of : ${error}`));
+						await confirmation.update({
+							content: `${targetMember} has been kicked for: ${reason}`,
+							components: []
+						}).catch(() => {});
+					} else if (confirmation.customId === 'cancel_user_kick') {
+						await confirmation.update({
+							content: 'Kick request cancelled',
+							components: []
+						});
+					}
+				} catch (e) {
+					await interaction.editReply({ content: 'Timeout waiting for confirmation', components: [], ephemeral: true });
+				}
+				break;
+			}
+			case 'ban': {
+				let userToBan = targetMember ?? interaction.options.getUser('user');
+				let reason = interaction.options.getString('reason') ?? "No reason provided";
+				let purgeDuration = interaction.options.getInteger('delete-messages') ?? 0

```

Comment:
Semicolon. Please run jshint and jsbeautify as noted in the git readme.

Reply:
Resolved in 7bbcdcc